### ISSUE TITLE
Provision to have a configurable ECR url 

### ISSUE THREAD

**Issue by rajikrishnamoorthy** [aws/copilot-cli#1629]
Copilot pushes the docker images to the existing account's ECR. But many a times the account where pipeline runs and the account where docker images are stored is different. If copilot can make the ECR url configurable it would help teams who have cross account ECR images.

**Comment by efekarakus** [comment]
Heya @rajikrishnamoorthy ! Copilot will initialize an ECR repository for each service in your application. 
If I'm understanding right, it seems you'd like to configure the ECR repository that Copilot pushes to. 
Is that right? Is there a reason why you cannot use the ECR repositories created with Copilot?

**Comment by rajikrishnamoorthy** [comment]
Hi @efekarakus, we have teams that maintain a central account that hosts ECR and the developers would push their docker images from their respective accounts. copilot would create ECR repo in the account where the CLI is configured for. But I would need to push the image to the central account. We have 2 accounts in this scenario, one account from where copilot init is run and the other account to whose ECR I need the image to be pushed.

**Comment by iamhopaul123** [comment]
Hello @rajikrishnamoorthy, in this situation one workaround for now I think is to manually build and push your image to that ECR repo then get the URL and specify the image URL in your manifest:

```
image:
  location: aws_account_id.dkr.ecr.region.amazonaws.com/my-svc:tag
```

**Comment by rajikrishnamoorthy** [comment]
Thanks @iamhopaul123 , the manifest.yml file gets created only after copilot init is run completely. So I would let copilot push the docker image to developer's aws account and then once the ECS is up and running, I manually push the image to the central ECR account, update the manifest.yml file with the image url and give copilot update. command Is my assumption correct?

**Comment by iamhopaul123** [comment]
Not really! You could run `svc init` to only create the manifest. Or you could choose not to deploy when you are doing `copilot init`.

So the typical workflow is: if you first use Copilot or you don't have much thing to configure, we recommend using `copilot init` to spin up your application easily. However, if you have some deployment configuration requirements, for example in your use case, you could run `app init` to create the application first then do `env init` to create an environment and `svc init` to create the service manifest. Once you get the service manifest, you could push your image to the centralized ECR repo and then modify the manifest. After that, just run `svc deploy` to deploy the service to any of those environments in the application.

For more information about Copilot, feel free to refer to our [website](https://aws.github.io/copilot-cli/)!

**Comment by rajikrishnamoorthy** [comment]
Thanks for the clarity @iamhopaul123 on _**env init**_ and _**app init**_. I was always sticking to _**copilot init**_ for creating the ECS. Now this helps.
But would making the ECR location a configurable parameter add value? Any idea of adding this as a feature?

**Comment by iamhopaul123** [comment]
Yeah I agree @rajikrishnamoorthy. It sounds like a good idea to make the push destination configurable for users. We could leave this issue open so as to keep you posted about it!

**Comment by rajikrishnamoorthy** [comment]
Great @iamhopaul123 , Thanks.

**Comment by bpottier** [comment]
For anyone else looking at this, this has been my solution for now: `yq e '.image.location = .image.location + ":" + env(VERSION)' -i copilot/<svc name>/manifest.yml` then a deploy.
